src/linker/ref/Linker/MessageContainer.cs

```diff
@@ -6,8 +6,8 @@ namespace Mono.Linker
 {
 	public readonly struct MessageContainer
 	{
-		public static MessageContainer CreateErrorMessage (string text, int code, string subcategory = "", MessageOrigin? origin = null, bool isWarnAsError = false) { throw null; }
-		public static MessageContainer CreateWarningMessage (LinkContext context, string text, int code, MessageOrigin origin, string subcategory = "") { throw null; }
+		public static MessageContainer CreateErrorMessage (string text, int code, string subcategory = "", MessageOrigin? origin = null, bool isWarnAsError = false, WarnVersion? version = null) { throw null; }
```

Comment:
This does not make sense to me `bool isWarnAsError = false, WarnVersion? version = null` is an implementation detail and should not be exposed to public API.